features/facebookImport/src/model/entities/message-threads-group.js

```diff
@@ -25,20 +25,28 @@ export default class MessageThreadsGroup {
     }
 
     forEachMessageThread(callback) {
-        for (const messageThread of this._messagesThreads) {
+        for (const messageThread of this.messagesThreads) {
             callback(messageThread);
         }
     }
 
     addMessageThreadFromData(messageThreadData) {
         const messageThread = new MessageThread();
         messageThread.initializeFromData(messageThreadData);
-        this._messagesThreads.push(messageThread);
+        const participant = messageThread.participants.toString();
+        if (this._messagesThreads[participant]) {
```

Comment:
Might be better here to use the `thread_path` as a key for identifying threads. That acts as an id of the thread. We could have for example two group threads with the same participants but diferent titles and messages.

Reply:
yes indeed that makes sense! Just fixed it :)